lib/Ast.ml

```diff
@@ -219,49 +245,112 @@ module Exp = struct
   let is_monadic_binding = test_id ~f:Longident.is_monadic_binding
 
   let is_symbol = test_id ~f:Longident.is_symbol
+
+  let is_sequence exp =
+    match exp.pexp_desc with
+    | Pexp_sequence _ -> true
+    | Pexp_extension
+        ( ext
+        , PStr
+            [ { pstr_desc=
+                  Pstr_eval (({pexp_desc= Pexp_sequence _; _} as e), [])
+              ; _ } ] )
+      when Source.extension_using_sugar ~name:ext ~payload:e ->
+        true
+    | _ -> false
+
+  let rec is_sugared_list' acc exp =
+    match exp.pexp_desc with
+    | Pexp_construct ({txt= Lident "[]"; _}, None) -> Ok (exp :: acc)
+    | Pexp_construct
+        ( {txt= Lident "::"; _}
+        , Some
+            { pexp_desc= Pexp_tuple [_; ({pexp_attributes= []; _} as tl)]
+            ; pexp_attributes= []
+            ; _ } ) ->
+        is_sugared_list' (exp :: acc) tl
+    | _ -> Error acc
+
+  let is_sugared_list =
+    let memo = Hashtbl.Poly.create () in
+    register_reset (fun () -> Hashtbl.clear memo) ;
+    fun exp ->
+      match Hashtbl.find memo exp with
+      | Some b -> b
+      | None -> (
+        match is_sugared_list' [] exp with
+        | Error l ->
+            List.iter ~f:(fun e -> Hashtbl.set memo ~key:e ~data:false) l ;
+            false
+        | Ok l ->
+            List.iter ~f:(fun e -> Hashtbl.set memo ~key:e ~data:true) l ;
+            true )
+
+  let is_simple (i, c) =
+    Poly.(c.Conf.module_item_spacing = `Compact)
+    && Location.is_single_line i.pexp_loc c.Conf.margin
+
+  let break_between _s ~cmts ~has_cmts_before ~has_cmts_after (i1, c1)
```

Comment:
it might be worth checking if this unused argument can be removed?

Reply:
good catch, done.